src/v2/Components/FlashBanner/__tests__/FlashBanner.jest.tsx

```diff
@@ -1,32 +1,246 @@
-    window.location = fakeLocation
+  it("renders based on a flash_message query param", () => {
+    window.location = { search: "?flash_message=expired_token" } as any
 
     const wrapper = mount(<FlashBanner />)
 
     expect(wrapper.text()).toContain("Link expired. Resend verification email.")
   })
+
+  it("returns nothing for an unsupported content code", () => {
+    const wrapper = mount(<FlashBanner contentCode="porntipsguzzardo" />)
+
+    expect(wrapper.find(Banner).exists()).toBeFalsy()
+  })
+
+  it("contentCode takes precedence over query string and props.me", () => {
+    window.location = { search: "?flash_message=expired_token" } as any
+
+    const wrapper = mount(
+      <FlashBanner
+        contentCode="invalid_token"
+        me={{
+          canRequestEmailConfirmation: true,
+        }}
+      />
+    )
+
+    expect(wrapper.text()).toContain("An error has occurred.")
+  })
+
+  it("query string takes precedence over props.me", () => {
+    window.location = { search: "?flash_message=expired_token" } as any
+
+    const wrapper = mount(
+      <FlashBanner
+        me={{
+          canRequestEmailConfirmation: true,
+        }}
+      />
+    )
+
+    expect(wrapper.text()).toContain("Link expired. Resend verification email.")
+  })
+})
+
+describe("Email Confirmation CTA", () => {
+  describe("user cannot request email confirmation", () => {
+    it("user is not prompted to request email confirmation if no logged in user is present", async () => {
+      const { wrapper } = await getRelayWrapper({ data: { me: null } })
+
+      expect(wrapper.find(Banner).exists()).toBeFalsy()
+    })
+
+    it("user is not prompted to request email confirmation if !me.canRequestEmailConfirmation", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: {
+          me: { id: "woot", canRequestEmailConfirmation: false },
+        },
+      })
+
+      expect(wrapper.find(Banner).exists()).toBeFalsy()
+    })
+  })
+
+  describe("user can request email confirmation (me?.canRequestEmailConfirmation=true)", () => {
+    it("user is prompted to request email confirmation if they can", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+    })
+
+    it("user seeing banner can click to trigger email confirmation message", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              unconfirmedEmail: "ceo@blackwater.biz",
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+
+      wrapper
+        .find("button")
+        .first()
+        .prop("onClick")({} as any)
```

Comment:
pretty minor: 

```suggestion
        .simulate('click')
```

also works, avoids an `any`, and is arguable a bit easier on the eyes